core/src/main/java/rdx/works/core/sargon/AuthorisedDappExtensions.kt

```diff
@@ -0,0 +1,123 @@
+package rdx.works.core.sargon
+
+import com.radixdlt.sargon.AuthorizedDapp
+import com.radixdlt.sargon.AuthorizedPersonaSimple
+import com.radixdlt.sargon.IdentityAddress
+import com.radixdlt.sargon.Persona
+import com.radixdlt.sargon.PersonaData
+import com.radixdlt.sargon.ProfileNetwork
+import com.radixdlt.sargon.ReferencesToAuthorizedPersonas
+import com.radixdlt.sargon.RequestedNumberQuantifier
+import com.radixdlt.sargon.RequestedQuantity
+import com.radixdlt.sargon.SharedPersonaData
+import com.radixdlt.sargon.SharedToDappWithPersonaAccountAddresses
+import com.radixdlt.sargon.Timestamp
+import com.radixdlt.sargon.extensions.getBy
+import com.radixdlt.sargon.extensions.init
+import com.radixdlt.sargon.extensions.invoke
+import com.radixdlt.sargon.extensions.updateOrAppend
+import rdx.works.core.mapWhen
+
+fun AuthorizedDapp.hasAuthorizedPersona(personaAddress: IdentityAddress): Boolean {
+    return referencesToAuthorizedPersonas.getBy(personaAddress) != null
+}
+
+fun AuthorizedDapp.updateAuthorizedDAppPersonas(
```

Comment:
this appends personas? name is a bit misleading if this is the case

Reply:
I used the same name as [before](https://github.com/radixdlt/babylon-wallet-android/blob/1591a86de906c6b10cbe87f057ebfe123a7c7f6d/profile/src/main/java/rdx/works/profile/data/repository/DAppConnectionRepository.kt#L282).

But before I do that can you explain me why do we need to prepend here?